docs/tutorials/lognamespacing.md

```diff
@@ -0,0 +1,229 @@
+fields try to have the same name. Log Namespacing does not apply to Metric events.
+
+## Config
+
+Add the following field to the `Config` struct:
+
+```rust
+    /// The namespace to use for logs. This overrides the global setting.
+    #[configurable(metadata(docs::hidden))]
+    #[serde(default)]
+    pub log_namespace: Option<bool>,
+```
+
+Currently, because log namespacing is an unreleased feature we add the `docs::hidden`
+metadata so it doesn't appear in the documentation.
+
+## Build
+
+The configuration is currently just a bool, switching it on or off. When we come to
+retrieve the actual namespace to use we merge it in with the globally configured one.
+This is passed in via the `SourceContext` parameter.
+
+```rust
+impl SourceConfig for DnstapConfig {
+    async fn build(&self, cx: SourceContext) -> Result<super::Source> {
+        let log_namespace = cx.log_namespace(self.log_namespace);
+```
+
+The `cx.log_namespace` function gives us a `LogNamespace` enum that we can use to
+set the fields in the appropriate section of the Event.
+
+This `log_namespace` variable needs to be passed to any functions that will insert 
+data into the log event that is emitted by the source.
+
+### Vector metadata
+
+The Vector namespace contains data pertinent to how the event was ingested into 
+Vector. Currently two fields need to be added to this namespace - `ingest_timestamp`
+and `source_type`:
+
+```rust
+  self.log_namespace.insert_vector_metadata(
+      &mut log_event,
+      path!(self.timestamp_key()),
+      path!("ingest_timestamp"),
+      chrono::Utc::now(),
+  );
+            
+  self.log_namespace.insert_vector_metadata(
+      &mut log_event,
+      path!(self.source_type_key()),
+      path!("source_type"),
+      DnstapConfig::NAME,
+  );
+```
+
+If we look at the parameters to `insert_vector_metadata`:
+
+```rust
+    pub fn insert_vector_metadata<'a>(
+        &self,
+        log: &mut LogEvent,
+        legacy_key: impl ValuePath<'a>,
+        metadata_key: impl ValuePath<'a>,
+        value: impl Into<Value>,
+    ) 
+```
+
+#### log
+
+This needs to be the log event that is being populated.
+
+#### legacy_key
+
+This is the name of the field the timestamp is to be inserted into
+when using the Legacy Namespace. This will typically be values returned
+by calls to `log_schema().source_type_key()` and `log_schema().timestamp_key()`.
+
+#### metdata_key
+
+The name of the field when it is inserted into the Vector namespace. This 
+will be `path!("ingest_timestamp")` or `path!(source_type)`. The field names
+can be hardcoded since they are going into the Vector namespace, so conflicts
+with other field names cannot occur.
+
+#### value
+
+The actual value to be placed into the field. 
+
+For the ingest timestamp this will be `chrono::Utc::now()`. Source type will be
+the `NAME` property of the `Config` struct. `NAME` is provided by the 
+`configurable_component` macro. You may need to include `use vector_config::NamedComponent;`.
```

Comment:
Note that batches of events can use a pre-calculated `Utc::now()` rather than calling for each event in the batch.